obsproject/obs-studio
# Various cleanups to linux-pipewire

### Description

A series of cleanups and code shuffling to prepare terrain for PipeWire-based cameras and audio.

### Motivation and Context

The current linux-pipewire code mixes the screencast portal code, with PipeWire streaming code. This makes it harder to introduce new PipeWire-based sources, such as cameras, virtual camera output, and audio.

### How Has This Been Tested?

 - Run this PR on a Linux / Wayland system
 - Observe that both monitor and window captures still work exactly as they used to

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:
